src/polycubed/src/polycubed_core.cpp

```diff
@@ -434,25 +423,33 @@ void PolycubedCore::detach(const std::string &cube_name,
 }
 
 std::string PolycubedCore::get_cube_port_parameter(
-    const std::string &cube, const std::string &port,
+    const std::string &cube, const std::string &port_name,
     const std::string &parameter) {
-  std::string service_name = ServiceController::get_cube_service(cube);
-  auto iter = servicectrls_map_.find(service_name);
-  if (iter == servicectrls_map_.end()) {
-    throw std::runtime_error("Service Controller does not exist");
-  }
+  Pistache::Http::Client client;
 
-  ServiceController &s = iter->second;
+  auto opts = Pistache::Http::Client::options();
+  client.init(opts);
+
+  auto port = std::to_string(config.getServerPort());
+  auto service_name = ServiceController::get_cube_service(cube);
+
+  std::string url("http://127.0.0.1:" + port + "/polycube/v1/" + service_name +
```

Comment:
Will it always be `127.0.0.1`?
Any use case, where the IP will be external/remote and we'll have to use it? I believe we can set the IP and Port as an environment variable, is it? 
Please correct me if I'm wrong. Thanks!

Reply:
Yes, it is a local connection because it is starting from the same process, so 127.0.0.1 should work in all cases. Anyway this code has to be improved in the future because this is not good to use tcp to access to the same process.

Reply:
This has already been improved, that static IP is not present in the code anymore.